Changelog.md

```diff
@@ -25,6 +26,7 @@ Bugfixes:
 
 
 AST Changes:
+ * AST: Add the ``experimentalParsingEnabled`` field to the ``SourceUnit`` nodes, which indicate whether the experimental parsing mode has been enabled via ``pragma experimental solidity``.
```

Comment:
```suggestion
 * AST: Add an optional ``experimental`` field to ``SourceUnit`` nodes, which is true if the AST stems from the experimental language mode enabled by ``pragma experimental solidity``.
```

Of course, this also means changing the name of the field and the flags in the rest of the PR, I won't comment on all of those. We can also make it ``experimentalSolidity``, if you prefer.
But:
- I'd really make it optional, i.e. the field is *only* present, when it's true.
- And parsing is really the least of the concerns in all this - the type system changes will be much more significant. Granted, the AST mainly stems from parsing, but it also contains type information after analysis, so the field name and its description here should reflect that.

Reply:
Actually, I think I like ``experimentalSolidity`` better than plain ``experimental`` myself :-).